### ISSUE TITLE
Alt key sometimes getting stuck after Alt+Tab

### ISSUE THREAD

**Issue by Saviq** [canonical/mir#3307]
We fixed this before:
- #2946

But it's biting me again from time to time. Of course no reproducer, but we may need to add some debug logging or something so we can track this down.

**Comment by Saviq** [comment]
Data point: it gets cleared after the first event.

**Comment by mattkae** [comment]
## Notes
Last time this was a problem, we were failing on `WlKeyboard::send_modifiers`. In this case, we were failing to send the modifiers due to a faulty condition. My suspicion here is that we're somehow sending incorrect modifiers, or not sending them at all.

**Comment by mattkae** [comment]
Let's get some logging first: https://github.com/canonical/mir/pull/3351

**Comment by AlanGriffiths** [comment]
I have a reproducer for what is likely the same bug: I have a script to swap my outputs (because they come up in an indeterminate order):
```
$ cat `which miriway-swap`
#!/bin/sh
exec sed -i "s/position: \[1920, 0\]/@@@/;s/position: \[0, 0\]/position: \[1920, 0\]/;s/@@@/position: \[0, 0\]/" ~/.config/miriway-shell.display
```
And have that bound:
```
$ grep swap ~/.config/miriway-shell.config
ctrl-alt=s:miriway-swap
```
I have found that after using `Ctrl-Alt-S`, I need to press and release `Alt` before it works again.

(Sorry, not running a build with your debug info yet, maybe later)

**Comment by mattkae** [comment]
@AlanGriffiths Is that reproducer consistent for you, or does it still happen only sometimes? I am unable to reproduce it on my end when I follow your steps

**Comment by AlanGriffiths** [comment]
> Is that reproducer consistent for you, or does it still happen only sometimes?

When I posted, it seemed consistent. Now, when I retry, not only it is intermittent, but pressing and releasing `Alt` doesn't fix it reliably either. Not sure what is different, but will monitor in the hope of enlightenment.

**Comment by mattkae** [comment]
> > Is that reproducer consistent for you, or does it still happen only sometimes?
> 
> When I posted, it seemed consistent. Now, when I retry, not only it is intermittent, but pressing and releasing `Alt` doesn't fix it reliably either. Not sure what is different, but will monitor in the hope of enlightenment.

Ok let me know! Remember that you can use this pull request to get some better info: https://github.com/canonical/mir/pull/3351

**Comment by AlanGriffiths** [comment]
Fixed by #3382
